public/js/visualisation_helpers.js

```diff
@@ -31,22 +31,45 @@ export function getPrefix(str) {
 }
 
 /**
- * Defines how ticks will be formatted.
+* Defines how ticks will be formatted.
  *
  * Examples: 200 aa, 2.4 kbp, 7.6 Mbp.
  *
  * Borrowed from Kablammo. Modified by Priyam based on https://github.com/mbostock/d3/issues/1722.
  */
 export function tick_formatter(scale, seq_type) {
-    var prefix = d3.format('~s')
-    var suffixes = {amino_acid: 'aa', nucleic_acid: 'bp'};
+    const prefix = d3.format('~s')
+    const suffixes = {amino_acid: 'aa', nucleic_acid: 'bp'};
 
     return function (d) {
-        return `${prefix(d)}${suffixes[seq_type]}`
-            .replace(/([a-zA-Z]+)/, ' $1')
+        const formatted = prefix(d);
+
+        return `${formatNumberUnits(formatted)}${suffixes[seq_type]}`;
     };
 }
 
+function extractSuffix(str) {
+    const suffix = str.replace(/[0-9.]/g, '');
+    const numericPart = d3.format('.2f')(parseFloat(str));
+
+    return [parseFloat(numericPart), suffix];
+}
+
+export function formatNumberUnits(number, threshold = 0.95) {
+    const [numericPart, suffix] = extractSuffix(number);
+    const integerPart = Math.floor(numericPart);
+    const fractionalPart = parseFloat((numericPart - integerPart).toFixed(2));
+
+    // when fractionalPart more than the threshold round up
+    // example:
+    //  threshold 0.95
+    //  2.95 -> 3
+    //  2.94 -> 2.9
+    const formatted = fractionalPart >= threshold ? integerPart + 1 : Math.floor(numericPart * 10) / 10;
```

Comment:
Do we need this custom rounding logic threshold? Am I right in thinking that we could instead round up to 1 decimal place, then remove ".0" at the end if it's a whole number? I think it would achieve the same result with a much easier to understand and simpler code, or am I missing something?

Reply:
@tadast the threshold based on suggested logic from Daniel

> divide these numbers by 1000 and round to the nearest decimal place
> 
> 19,299 bp = 19.3 kbp
> 19,990 bp = 20.0 kbp
> divide these numbers by 1000000 and round to the nearest decimal place
> 
> 199,929,999 bp = 199.9 Mbp
> 199,990,000 bp = 200.0 Mbp

Reply:
@tadast I updated the code, it's more simple now